Tasks/google-play-release/GooglePlay.js

```diff
@@ -227,29 +239,400 @@ function updateTrack(packageName, track, versionCode, userFraction) {
+
+    try {
+        var changelogs = fs.readdirSync(changelogDir).filter(function (subPath) {
+            var pathIsFile = false;
+            try {
+                var fileToCheck = path.join(changelogDir, subPath);
+                tl.debug(`Checking File ${fileToCheck}`);
+                pathIsFile = fs.statSync(fileToCheck).isFile();
+            } catch (e) {
+                tl.debug(e);
+                tl.debug(`failed to stat path ${subPath}. ignoring...`);
+            }
+
+            return pathIsFile;
+        });
+
+        for (var i in changelogs) {
+            var fullChangelogPath = path.join(changelogDir, changelogs[i]);
+            addAllChangelogsPromise = addAllChangelogsPromise.then(function (changelog) {
+                console.log(`Appending changelog ${changelog}`);
+                return addChangelog.bind(this, languageCode, changelog)();
+            }.bind(this, fullChangelogPath));
+        }
+    } catch (e) {
+        tl.debug(e);
+        tl.debug(`no changelogs found in ${changelogDir}`);
+    }
+
+    return addAllChangelogsPromise;
+}
+
+/**
+ * Attaches the metadata in the specified directory to the edit. Assumes the metadata structure specified by Fastlane.
+ * Assumes authorized
+ * 
+ * Metadata Structure:
+ * metadata
+ *  └ $(languageCodes)
+ *    ├ full_description.txt
+ *    ├ short_description.txt
+ *    ├ title.txt
+ *    ├ video.txt
+ *    ├ images
+ *    |  ├ featureGraphic.png    || featureGraphic.jpg   || featureGraphic.jpeg
+ *    |  ├ icon.png              || icon.jpg             || icon.jpeg
+ *    |  ├ promoGraphic.png      || promoGraphic.jpg     || promoGraphic.jpeg
+ *    |  ├ tvBanner.png          || tvBanner.jpg         || tvBanner.jpeg
+ *    |  ├ phoneScreenshots
+ *    |  |  └ *.png || *.jpg || *.jpeg
+ *    |  ├ sevenInchScreenshots
+ *    |  |  └ *.png || *.jpg || *.jpeg
+ *    |  ├ tenInchScreenshots
+ *    |  |  └ *.png || *.jpg || *.jpeg
+ *    |  ├ tvScreenshots
+ *    |  |  └ *.png || *.jpg || *.jpeg
+ *    |  └ wearScreenshots
+ *    |     └ *.png || *.jpg || *.jpeg
+ *    └ changelogs
+ *      └ $(versioncodes).txt
+ * 
+ * @param {string} metadataDirectory Path to the folder where the Fastlane metadata structure is found
+ * @returns {Promise}  A promise that will return the result from last metadata change that was attempted. Currently, this is most likely an image upload.
+ *                     { image: { id: string, url: string, sha1: string } }
+ */
+function addMetadata(metadataDirectory) {
+    tl.debug("Attempting to add metadata...");
+    metadataDirectory = path.resolve(metadataDirectory);
+    var metadataRootDirectory = path.join(metadataDirectory, "metadata");
+    tl.debug(`Assuming root is at ${metadataDirectory}\n\rAdding metadata from ${metadataRootDirectory}`);
+
+    var metadataLanguageCodes = fs.readdirSync(metadataRootDirectory).filter(function (subPath) {
```

Comment:
Discussed offline. For information:
Under the metadata folder is a bunch of languageCode folders, and inside each language code folder is the structure listed above. We don't do any verification of the folder names, and the task will fail if the server rejects one of the language code names.